### DISCUSSION TITLE
Nominal Level-2 Predictor Imputation

### DISCUSSION THREAD

**Discussion by janlouwkotze** [amices/mice#643] [Missing data methodology]
Hello!

I am running a multilevel model that has continuous, dichotomous, and nominal predictor variables at both level 1 and 2. I have been able to identify relevant imputation methods for the dichotomous and continuous variables at level 1 and 2 (e.g., "2l.norm" and "logreg"). I also identified the relevant imputation method for level 1 nominal variables (i.e., "polyreg"). However, I can't seem to find an imputation function that can handle level 2 nominal predictors. Is anyone aware of where I can this function?

My understanding is that I could use "2lonly.function" to impute nominal level-2 variables if I specify, for example, "polyreg" for "imputationFunction." However, when I do this and run the imputation model, I receive multiple warnings that read as such, "Warning: glmer does not run. Simplify imputation model." This is perplexing because the model appears to run (in spite of the warnings) and produces a fully imputed dataset. Is it acceptable to use this dataset, despite the warnings messages?

**Comment by stefvanbuuren** [comment]
Not quite clear which imputation method you specify. In `mice` only "2l.bin" will generate this message. Please try to provide a reproducible example to help the helpers.

**Reply by janlouwkotze** [reply]
Thank you for your quick reply, Stef!
Based on your comment, I managed to figure out the issue with the warnings by substituting the "2l.bin" estimator.
Have a nice day.

**Reply by stefvanbuuren** [reply]
OK, but that still does not tell me what you did in the first place. 

As you managed to find a solution, I am going to close now. Feel free to reopen if needed.
